### ISSUE TITLE
About separation convolution implementation

### ISSUE THREAD

**Issue by hhaAndroid** [ThomasZiegler/Efficient-Smoothing-of-Dilated-Convolutions#1]
hello:I looked at the code implementation and have a question. 
_averaged_dilated_conv2d and _gaussian_dilated_conv2d_fix are similar in principle, but why one uses separate convolution and one uses 3d convolution，Can't you all use depthwise_conv2d_native?I don't know why.Thank you

**Comment by ThomasZiegler** [comment]
Hi
Yes your are right *depthwise_conv2d_native* should also work with the*_gaussian_dilated_conv2d_fix* filter. We actually did at one point have this method also for the Gaussian filters. The the change was in hinge side not required. We had problem to get a gradient on the sigma parameter hence we changed to 3d conv (this was how the original code implemented the separate convolution). At the end the problem was related to the variable scope, hence the ugly *top_scope* variable in each method.
This whole project was part of a course at ETH and at this point we did already run a lot of experiments with the 3d conv method so we kept the code the way we got our results.
